R/between_within_classic.R

```diff
@@ -0,0 +1,98 @@
+#' Calculation of between-within Degrees of Freedom for One-Dimensional Contrast
+#'
+#' @description Calculates the estimate, standard error, degrees of freedom,
+#' t statistic and p-value for one-dimensional contrast. Used in [df_1d()] if method is
+#' "Between-within".
+#'
+#' @param object (`mmrm`)\cr the MMRM fit.
+#' @param contrast (`numeric`)\cr contrast vector. Note that this should not include
+#'   elements for singular coefficient estimates, i.e. only refer to the
+#'   actually estimated coefficients.
+#'
+#' @return List with `est`, `se`, `df`, `t_stat` and `p_val`.
+#' @keywords internal
+h_df_1d_bwc <- function(object, contrast) {
```

Comment:
the code looks very close to `h_df_1d_bw`. not sure what is the best path but can we avoid more the code duplication? e.g. how about giving another argument that switches the code between `standard` and `classic` behavior? (I guess classic is SAS similar?..)